include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -4418,6 +4421,666 @@ namespace internal
+    }
+
+    template <int          fe_degree,
+              unsigned int direction,
+              unsigned int side,
+              bool         transpose>
+    static void
+    interpolate_3D_face(const unsigned int dof_offset,
+                        const unsigned int given_degree,
+                        bool               is_subface_0,
+                        const unsigned int v,
+                        const Number *     weight,
+                        Number *           values)
+    {
+      if (is_subface_0)
+        interpolate_3D_face<fe_degree, direction, side, transpose, true>(
+          dof_offset, given_degree, v, weight, values);
+      else
+        interpolate_3D_face<fe_degree, direction, side, transpose, false>(
+          dof_offset, given_degree, v, weight, values);
+    }
+
+    template <int          fe_degree,
+              unsigned int direction,
+              unsigned int side,
+              bool         transpose,
+              bool         is_subface_0>
+    static void
+    interpolate_3D_face(const unsigned int dof_offset,
+                        const unsigned int given_degree,
+                        const unsigned int v,
+                        const Number *     weight,
+                        Number *           values)
+    {
+      typename Number::value_type temp[10 /*TODO*/];
+
+      const unsigned int points =
+        (fe_degree != -1 ? fe_degree : given_degree) + 1;
+      const unsigned int stride = Utilities::pow(points, direction);
+      const unsigned int d      = side / 2;
+
+      // direction   side0   side1   side2
+      // 0             -      p^2      p
+      // 1            p^2      -       1
+      // 2             p       -       1
+      const unsigned int stride2 =
+        ((direction == 0 && d == 1) || (direction == 1 && d == 0)) ?
+          (points * points) :
+          (((direction == 0 && d == 2) || (direction == 2 && d == 0)) ? points :
+                                                                        1);
+
+      for (unsigned int g = 1; g < points - 1; ++g)
+        {
+          // copy result back
+          for (unsigned int k = 0; k < points; ++k)
+            temp[k] = values[dof_offset + k * stride + stride2 * g][v];
+
+          // perform interpolation point by point
+          for (unsigned int k = 0; k < points; ++k)
+            for (unsigned int h = 0; h < points; ++h)
+              {
+                const unsigned int index_v =
+                  dof_offset + k * stride + stride2 * g;
+                const unsigned int index_w =
+                  ((transpose ? 1 : points) *
+                   (is_subface_0 ? k : (points - 1 - k))) +
+                  ((transpose ? points : 1) *
+                   (is_subface_0 ? h : (points - 1 - h)));
+                if (h == 0)
+                  values[index_v][v] = weight[index_w][v] * temp[h];
+                else
+                  values[index_v][v] += weight[index_w][v] * temp[h];
```

Comment:
Here I believe the same will happen, i.e., the compiler does too many stores on `values`.